packages/indy-vdr/src/anoncreds/IndyVdrAnonCredsRegistry.ts

```diff
@@ -552,48 +582,211 @@ export class IndyVdrAnonCredsRegistry implements AnonCredsRegistry {
     }
   }
 
-  public async registerRevocationRegistryDefinition(): Promise<RegisterRevocationRegistryDefinitionReturn> {
-    throw new AriesFrameworkError('Not implemented!')
-  }
-
-  public async getRevocationStatusList(
+  public async registerRevocationRegistryDefinition(
     agentContext: AgentContext,
-    revocationRegistryId: string,
-    timestamp: number
-  ): Promise<GetRevocationStatusListReturn> {
+    { options, revocationRegistryDefinition }: IndyVdrRegisterRevocationRegistryDefinition
+  ): Promise<IndyVdrRegisterRevocationRegistryDefinitionReturn> {
     try {
-      const indySdkPoolService = agentContext.dependencyManager.resolve(IndyVdrPoolService)
-
-      const { did, namespaceIdentifier, schemaSeqNo, credentialDefinitionTag, revocationRegistryTag } =
-        parseIndyRevocationRegistryId(revocationRegistryId)
-      const { pool } = await indySdkPoolService.getPoolForDid(agentContext, did)
+      // This will throw an error if trying to register a credential definition with a legacy indy identifier. We only support did:indy
+      // identifiers for registering, that will allow us to extract the namespace and means all stored records will use did:indy identifiers.
+      const { namespaceIdentifier, namespace } = parseIndyDid(revocationRegistryDefinition.issuerId)
+      const indyVdrPoolService = agentContext.dependencyManager.resolve(IndyVdrPoolService)
+      const pool = indyVdrPoolService.getPoolForNamespace(namespace)
 
       agentContext.config.logger.debug(
-        `Using ledger '${pool.indyNamespace}' to retrieve revocation registry deltas with revocation registry definition id '${revocationRegistryId}' until ${timestamp}`
+        `Registering revocation registry definition on ledger '${namespace}' with did '${revocationRegistryDefinition.issuerId}'`,
+        revocationRegistryDefinition
       )
 
-      const legacyRevocationRegistryId = getUnqualifiedRevocationRegistryId(
-        namespaceIdentifier,
-        schemaSeqNo,
-        credentialDefinitionTag,
-        revocationRegistryTag
-      )
-      const request = new GetRevocationRegistryDeltaRequest({
-        revocationRegistryId: legacyRevocationRegistryId,
-        toTs: timestamp,
-      })
+      let writeRequest: CustomRequest
+      let didIndyRevocationRegistryDefinitionId: string
 
-      agentContext.config.logger.trace(
-        `Submitting get revocation registry delta request for revocation registry '${revocationRegistryId}' to ledger`
+      // TODO: this will bypass caching if done on a higher level.
+      const { credentialDefinition, resolutionMetadata } = await this.getCredentialDefinition(
```

Comment:
Why do we need to fetch the cred def? We can extract all needed parts from the cred def id. And indy will reject if something is wrong